src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+"""Calculate photon incident angles on focal plane and primary/secondary mirrors.
+
+Parses the imaging list (``.lis``) produced by sim_telarray_debug_trace and uses
+Angle of incidence at focal surface, with respect to the optical axis [deg],
+Angle of incidence on(to) primary mirror [deg], and
+Angle of incidence on(to) secondary mirror [deg] (if available).
+"""
+
+import logging
+import re
+import subprocess
+from pathlib import Path
+
+import astropy.units as u
+from astropy.table import QTable
+
+from simtools.model.model_utils import initialize_simulation_models
+
+
+class IncidentAnglesCalculator:
+    """Run a PSF-style sim_telarray job and compute incident angles at mirrors or focal surfaces.
+
+    Parameters
+    ----------
+    simtel_path : str or pathlib.Path
+        Path to the sim_telarray installation directory (containing ``sim_telarray/bin``).
+    db_config : dict
+        Database configuration passed to ``initialize_simulation_models``.
+    config_data : dict
+        Simulation configuration (e.g. ``site``, ``telescope``, ``model_version``,
+        ``off_axis_angle``, ``source_distance``, ``number_of_photons``).
+    output_dir : str or pathlib.Path
+        Output directory where logs, scripts, photons files and results are written.
+    label : str, optional
+        Label used to name outputs; defaults to ``incident_angles_<telescope>`` when omitted.
+
+    Notes
+    -----
+    Additional options are read from ``config_data`` when present:
+    - ``perfect_mirror`` (bool, default False)
+    - ``mirror_reflection_random_angle`` (float, optional)
+    - ``mirror_alignment_random`` (float, optional)
+    - ``calculate_primary_secondary_angles`` (bool, default True)
+    """
+
+    # Use fixed zenith angle (degrees) for incident-angle simulations.
+    ZENITH_ANGLE_DEG = 45.0
```

Comment:
Why did you choose 45 deg?

It is neither the simplest case (0 deg) or the average case (60-70 deg).

Reply:
It is simply not important as you said, but I can change it to any angle you wish.